### ISSUE TITLE
amp-autocomplete: Support `disabled` and `value` on arbitrary element items. 

### ISSUE THREAD

**Issue by caroqliu** [ampproject/amphtml#22133]
## Describe the new feature or change to an existing feature you'd like to see

At the moment, the way to disable an item on `amp-autocomplete` is by providing a `data-disabled` attribute on the `template` child of the component. Likewise, the way to provide the `value` of an item is to specify a `data-value` attribute. Instead, the developer should be able to specify this with first class `disabled` and `value` attributes. 

This will require validation changes similar to those that `amp-selector` does [here](https://github.com/ampproject/amphtml/blob/e9acfc63ccde5afe4d048d275b4320a771443396/extensions/amp-selector/validator-amp-selector.protoascii#L83-L126), where `option` and `selected` in the linked example could roughly be replaced with `value` and `disabled`.

## Describe alternatives you've considered

#22094 Allows disable functionality with a `data-disabled` attribute in the mean time.

**Comment by caroqliu** [comment]
Closing this issue because `value` and `disabled` are not compatible with arbitrary elements.

See [Errors that involve areas reserved for future expansion](https://html.spec.whatwg.org/multipage/introduction.html#restrictions-on-content-models-and-on-attribute-values)
